### PR TITLE
EDU-490 Compare email adresses in lower case

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-490

### CODE REVIEW

File: src/stores/user-store.js [hunk 1]
```diff
@@ -12,10 +12,11 @@ class UserStore {
   }
 
   async findUserByLogin({ provider, emailOrUsername }, { session } = {}) {
+    const lowerCasedEmailOrUsername = emailOrUsername?.toLowerCase() || '';
```
Comment (ahelmberger) [inline src/stores/user-store.js:15]: I realized this function has much more logic than the store should have (also with the `switch` below). I left it for the fix, but will create a ticket to put the decision logic into the service.

Comment (adrianaluchian) [reply src/stores/user-store.js:15]: That's true, the best match decision does not belong here. Actually this method should rather be "getUserByEmailOrUsername" .

**Review by adrianaluchian — APPROVED** [review]

File: src/stores/user-store.js (hunk 1 above)
Comment (adrianaluchian) [reply src/stores/user-store.js:15]: Oh wait, but for username we allow upper case 🤔

Comment (ahelmberger) [reply src/stores/user-store.js:15]: :+1:

Comment (adrianaluchian) [reply src/stores/user-store.js:15]: Don't wait, I see you only used it for email 😆 sorry!

**Pull request merged**